src/composables/useGetMessages.ts

```diff
@@ -0,0 +1,336 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import type { AxiosError } from '@nextcloud/axios'
+import type {
+	ComputedRef,
+	InjectionKey,
+	Ref,
+} from 'vue'
+import type {
+	ChatMessage,
+	Conversation,
+} from '../types/index.ts'
+
+import Axios from '@nextcloud/axios'
+import { subscribe, unsubscribe } from '@nextcloud/event-bus'
+import { computed, inject, onBeforeUnmount, provide, ref, watch } from 'vue'
+import { useRoute } from 'vue-router'
+import { useStore } from 'vuex'
+import { CHAT, MESSAGE } from '../constants.ts'
+import { debugTimer } from '../utils/debugTimer.ts'
+import { useGetThreadId } from './useGetThreadId.ts'
+import { useGetToken } from './useGetToken.ts'
+
+const loadingOldMessagesKey = Symbol.for('loadingOldMessages') as InjectionKey<Ref<boolean>>
+const isInitialisingMessagesKey = Symbol.for('isInitialisingMessages') as InjectionKey<Ref<boolean>>
+const stopFetchingOldMessagesKey = Symbol.for('stopFetchingOldMessages') as InjectionKey<Ref<boolean>>
+const isChatBeginningReachedKey = Symbol.for('isChatBeginningReached') as InjectionKey<ComputedRef<boolean>>
+
+const getMessageContextKey = Symbol.for('getMessageContext') as InjectionKey<(token: string, messageId: number) => Promise<void>>
+const getOldMessagesKey = Symbol.for('getOldMessages') as InjectionKey<(token: string, includeLastKnown: boolean) => Promise<void>>
+
+/**
+ * Check whether caught error is from OCS API
+ */
+function isAxiosErrorResponse(exception: unknown): exception is AxiosError<string> {
+	return typeof exception === 'object' && 'response' in exception!
```

Comment:
If `exception` is guaranteed to be non-nullable, it should be `NonNullable<unknown>`, not unknown.
If it isn't guaranteed, `!` should not be used (because it may be `null`).

```suggestion
	return exception !== null && typeof exception === 'object' && 'response' in exception
```

Reply:
`catch(exception` has type unknown and currently there is no other thing to narrow it down in Typescript:

> Vue: Catch clause variable type annotation must be any or unknown if specified